### ISSUE TITLE
URL for user profile encoded wrong if user name has special characters

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#6155]
##### Detailed steps causing the problem:
- Open GC34V8W
- Search the log of user `●•C•●` dated 13.11.16
- Click on the user name
- Select "Open profile in browser"

##### Actual behavior after performing these steps:
Browser opens with the following URL (which returns 404):
`https://www.geocaching.com/profile/?u=%E2%97%8F%E2%80%A2C%E2%80%A2%E2%97%8F`

##### Expected behavior after performing these steps:
Browser should open the correct URL

##### Version of c:geo used:
2016.11.05

##### Is the problem reproducible for you?
Yes

##### Other comments and remarks:
From support mail and confirmed

**Comment by SammysHP** [comment]
Actually that URL is correct.

**Comment by Lineflyer** [comment]
But giving me a 404 in my phone browser

**Comment by SammysHP** [comment]
Because gc.com cannot handle these characters. Open the page with his finds and click on the "Profile" link at the top: https://www.geocaching.com/seek/nearest.aspx?ul=%E2%97%8F%E2%80%A2C%E2%80%A2%E2%97%8F

**Comment by rsudev** [comment]
Yep. It is correct in the sense, that it encodes what we know (the username) in the right way to hand it over as a parameter to the query url.
Unfortunately does it look like this query is not capable of handling this parameter correctly (while it can in others)
Not sure if might have access to the user guid to be able to address the user successfully.

**Comment by SammysHP** [comment]
Who was last in contact with Groundspeak? Maybe we should report it to them.

**Comment by samueltardieu** [comment]
It works for me. Using `curl` I can retrieve the page and save it, then it shows properly in my browser and is the user page.

**Comment by samueltardieu** [comment]
To be precise: the URL from @SammysHP works on me (`/seek/nearest.aspx`), not the one from @Lineflyer.

**Comment by SammysHP** [comment]
@samueltardieu Yes, that is what I said. The URL is correct and is even used by Groundspeak in `/seek/nearest.aspx`. So they handle it correctly in some cases, but not in others like `/profile/`.

**Comment by Lineflyer** [comment]
Thanks for your explanation. I also think in this case we should leave it like that unless we have any chance to query with the users `guid` instead.
Maybe the will at some time recognize the `404` errors and fix.
I wonder if they use this function theirselves on some link/page? I was unable to find it yet.

**Comment by Lineflyer** [comment]
Addon:
We should already parse `https://www.geocaching.com/seek/cache_logbook.aspx?guid=5a1b3e10-c159-41b4-a32d-37eba27063be` and this page contains the guid of the users in the logbook. Can't we use that instead?

**Comment by samueltardieu** [comment]
I prefer that we use the extra request needed to get the GUID as needed rather than store a new field specific to geocaching.com.

**Comment by SammysHP** [comment]
> I wonder if they use this function theirselves on some link/page? I was unable to find it yet.

Please read my comment above: https://github.com/cgeo/cgeo/issues/6155#issuecomment-262316621

**Comment by samueltardieu** [comment]
> Open the page with his finds and click on the "Profile" link at the top: https://www.geocaching.com/seek/nearest.aspx?ul=%E2%97%8F%E2%80%A2C%E2%80%A2%E2%97%8F

I have no such "Profile" link.

**Comment by SammysHP** [comment]
Then it might be an addition from GC Little Helper. I'll check later when I'm at home.

**Comment by Lineflyer** [comment]
@SammysHP Did you have the chance to check?

**Comment by SammysHP** [comment]
@Lineflyer Yes, sorry for the late reply. The link was added by GC Little Helper. Still it's a problem of gc.com.

**Comment by Lineflyer** [comment]
So, closing.

**Comment by SammysHP** [comment]
But someone should write a message to Groundspeak and report the problem.

**Comment by Lineflyer** [comment]
I still have the ToDo to contact them and announce our "official" contact mail to them. I can do that alongside.
